### PR TITLE
Purchases: Update manage purchase design

### PR DESCRIPTION
**Pull request by ryelle**
See #12746, implementing a new design for the Purchases section. This PR implements the new design for the main area of the single purchase view.

screen shot 2017-05-01 at 1 43 13 pm

There have been no functionality changes, only added data to this page to display the plan info, slight code restructuring, and some linting cleanup.

To test:

1. Visit `/me/purchases`
2. Check a wp.com plan purchase
3. Check a Jetpack plan purchase
4. Check a domain mapping purchase

And any other purchases you may have on your account

### CODE REVIEW

**Comment by rickybanister** [comment]
This looks great. I noticed a small improvement we can make, but will open a separate ticket.

File: client/me/purchases/manage-purchase/index.jsx [hunk 1]
```diff
@@ -254,98 +269,145 @@ const ManagePurchase = React.createClass( {
 				{ translate( 'Cancel Privacy Protection' ) }
 			</CompactCard>
 		);
-	},
+	}
 
-	renderPurchaseDetail() {
```
Comment (tyxla) [inline client/me/purchases/manage-purchase/index.jsx]: With these changes we might be losing substantial information about some purchase types. For example, if you purchased a theme, in the detail page you might not see that it's actually a theme:



while previously it was indicating that it was a premium theme and it included a link to the theme page:



We should make sure to check that with all purchase types.

File: client/me/purchases/manage-purchase/index.jsx [hunk 2]
```diff
@@ -142,13 +153,17 @@ const ManagePurchase = React.createClass( {
...
 		page( '/checkout/' + this.props.selectedSite.slug );
-	},
+	}
 
 	renderRenewButton() {
 		const purchase = getPurchase( this.props );
 		const { translate } = this.props;
 
-		if ( ! config.isEnabled( 'upgrades/checkout' ) || ! isRenewable( purchase ) || isExpired( purchase ) || isExpiring( purchase ) || ! getSelectedSite( this.props ) ) {
+		if ( ! config.isEnabled( 'upgrades/checkout' ) ) {
```
Comment (johnHackworth) [inline client/me/purchases/manage-purchase/index.jsx]: why did you divide this condition into two different ones?

Comment (tyxla) [inline client/me/purchases/manage-purchase/index.jsx]: Maybe we can remove that check? It's currently enabled for all environments.

Comment (ryelle) [inline client/me/purchases/manage-purchase/index.jsx]: line length 😣  it seemed clearer than breaking the conditions into two lines

File: client/me/purchases/manage-purchase/index.jsx [hunk 3]
```diff
@@ -254,98 +269,145 @@ const ManagePurchase = React.createClass( {
...
-			siteName,
-			siteDomain,
-			productLink,
-			renewButton,
-			editPaymentMethodNavItem,
-			cancelPurchaseNavItem,
-			cancelPrivacyProtectionNavItem;
+	renderPlanIcon() {
+		const purchase = getPurchase( this.props );
+		if ( isPlan( purchase ) ) {
```
Comment (tyxla) [inline client/me/purchases/manage-purchase/index.jsx]: Do we intend to add icons for more purchase types?

**Comment by johnHackworth** [comment]
I left a small comment, but this looks good to me :+1:

File: client/me/purchases/manage-purchase/index.jsx [hunk 4]
```diff
@@ -142,13 +153,17 @@ const ManagePurchase = React.createClass( {
...
 	renderRenewButton() {
 		const purchase = getPurchase( this.props );
 		const { translate } = this.props;
 
-		if ( ! config.isEnabled( 'upgrades/checkout' ) || ! isRenewable( purchase ) || isExpired( purchase ) || isExpiring( purchase ) || ! getSelectedSite( this.props ) ) {
+		if ( ! config.isEnabled( 'upgrades/checkout' ) ) {
+			return null;
+		}
+
+		if ( ! isRenewable( purchase ) || isExpired( purchase ) || isExpiring( purchase ) || ! getSelectedSite( this.props ) ) {
```
Comment (ryelle) [inline client/me/purchases/manage-purchase/index.jsx]: Sure, but this section already uses these utilities, and I didn't want to add to this PR by removing them - like you said, out of scope for this PR.

File: client/me/purchases/manage-purchase/index.jsx (hunk 2 above)
Comment (johnHackworth) [inline client/me/purchases/manage-purchase/index.jsx]: I wouldn't do it ... at least in this PR :)

I mean, we probably should do it, but we shouldn't remove a single case in a separate PR, but doing a different branch that removes all the use cases at once

File: client/me/purchases/manage-purchase/index.jsx (hunk 1 above)
Comment (ryelle) [inline client/me/purchases/manage-purchase/index.jsx]: Good catch on the theme details — @rickybanister what do you think?

**Comment by rickybanister** [comment]
Glad you caught that @tyxla. Let's do a couple things here:

**Domain Purchase**

* Add a descriptor below the title. Example for a domain purchase with privacy protection
* Add a clickable card for editing domain details

image

image

**Theme Purchase**

* Add an icon for theme purchases. We can use `<Gridicon icon="themes" />` and use 8px rounded corners on the parent element, not the gridicon itself (using a circle cropped the theme icon).
* Use the plan purchase's 'view plan details' link pattern for 'theme details'

image

**Comment by ryelle** [comment]
@rickybanister What do you think of using the plans layout for this related info, like this:

screen shot 2017-05-04 at 1 18 50 pm

**Comment by ryelle** [comment]
I swapped the PR labels back to needs review/design review because of https://github.com/Automattic/wp-calypso/pull/13533#issuecomment-299048421, but nothing has changed here yet.

**Comment by rickybanister** [comment]
Can we pull theme descriptions to populate that paragraph area? I'm ok with using the plans layout if it won't look blank—the colored border is a bit much without any kind of description there.

**Comment by ryelle** [comment]
screen shot 2017-05-04 at 6 13 07 pm

How's this? Some themes might have only 1-sentence short descriptions, but I think most are about this length.

**Comment by rickybanister** [comment]
@ryelle I think that's great, can we still sneak in the small text that mentions 'domain + private registration' or 'premium theme' under the name and above the price? I think that would help as well. But your solution to use the plan header seems great.

**Comment by ryelle** [comment]
Sure, I can do that on theme purchases, but it would be repetitive info for domain purchases, see https://github.com/Automattic/wp-calypso/pull/13533#issuecomment-299251681 -- the `.blog Domain Registration` would appear twice (in the header, and as the description)

**Comment by rickybanister** [comment]
Perhaps we can ask someone from Spectrum to help contribute a longer description or bit of marketing copy for domain purchases? The short descriptor is helpful for scanning, and the longer one can show the value of the domain or other upgrade?

cc @ranh @breezyskies @michelleweber

**Comment by ranh** [comment]
Thanks for the ping @rickybanister 

This is a simple description we've used in the past for domains: 

> Replaces your site's free address (example.wordpress.com) with the domain example.com, making it easier to remember and easier to share.

You may want to check with Marketing about more recent efforts along those lines.

Unrelated note, the use of one-word plan names here feels quite jarring.

https://cloud.githubusercontent.com/assets/541093/25587933/378b5bb0-2e74-11e7-9def-73f3f3c38c1e.png

"Professional" is not a product name. 

It stands out both in the title (next to the price) and in the expiration notice, because it feels like an incomplete sentence: an adjective without a noun. 

As recommended by the style guide, I think the full "Jetpack Professional" would be much more appropriate here.

**Comment by ryelle** [comment]
Hm, the purchase object itself only has it as "Professional" (vs the WP.com plans which do say "WordPress.com Business") and this goes back to wp.com code. If this is something that should be consistently done, maybe it should be changed there?

For now, I've added a "Jetpack" prefix to all Jetpack plan names, but I'm not sure if/how this should be handled for translation. Really the ideal would be to add Jetpack to the plan names on wp.com, and they're individually translated. I've also added the purchase type to the header.

screen shot 2017-05-08 at 1 56 26 pm

screen shot 2017-05-08 at 1 55 51 pm

@ranh Do you know if that text is already displayed anywhere in calypso? The only thing I can find like it is the "Free Domain" tooltip text on the plans page, but that's not really the same thing. Also, do you also have text for domain registration?

File: client/me/purchases/manage-purchase/index.jsx [hunk 5]
```diff
@@ -41,19 +43,32 @@ import {
...
 	getSelectedSite as getSelectedSiteSelector,
 	getSelectedSiteId,
 } from 'state/ui/selectors';
+import Gridicon from 'gridicons';
 import HeaderCake from 'components/header-cake';
-import { isDomainRegistration } from 'lib/products-values';
+import {
+	isPersonal,
+	isPremium,
+	isBusiness,
```
Comment (sirreal) [inline client/me/purchases/manage-purchase/index.jsx]: Nit: It would be nice to alphabetize these 🙂

**Comment by rickybanister** [comment]
@ranh yes, I agree about the plan names, and as @ryelle mentions, that's something that needs to happen in the billing system (or something). 

@ryelle do you think you could wrangle someone else from Luna to handle that as you transition to your new role? I think it does make sense to make sure that all happens in the right place.

**Comment by ranh** [comment]
> Do you know if that text is already displayed anywhere in calypso? The only thing I can find like it is the "Free Domain" tooltip text on the plans page, but that's not really the same thing. Also, do you also have text for domain registration

I don't know if this copy is used elsewhere in Calypso, probably not. But I also don't know if we have any other copy to use instead. I know that Marketing are working on domains ATM, so maybe they'll have something better to suggest.

> Hm, the purchase object itself only has it as "Professional" (vs the WP.com plans which do say "WordPress.com Business") and this goes back to wp.com code. If this is something that should be consistently done, maybe it should be changed there?

> @ranh yes, I agree about the plan names, and as @ryelle mentions, that's something that needs to happen in the billing system (or something).

That sounds right. I mentioned it here because this is the first I'm seeing this, but if it's a larger issue I completely agree that it should be handled at the root.

**Comment by richardmuscat** [comment]
@rickybanister / @ranh / @ryelle / @johnHackworth -- absolutely agree, we should start labelling them as soon as feasible as:
- Jetpack Personal
- Jetpack Premium
- Jetpack Professional

(As background, the word Jetpack is not their currently because when we kicked these off we didn't, at the time, think we wanted the Jetpack brand to be prominently mentioned on WordPress.com. As you know, that decision has now been reversed.)

**Comment by johnHackworth** [comment]
I'm not sure about the implications of changing the plans names. There shouldn't be any, but you know ...  I rather prefer if we merge this PR with @ryelle's fix and then I can take a look at it globally in a different branch

**Comment by rickybanister** [comment]
@johnHackworth agreed. If we can sort out the copy issue mentioned above then this is ready to merge. I'd also be ok with introducing a new string for this.

**Comment by johnHackworth** [comment]
I think we can ship this as soon as the copy issue mentioned by rick up there is fixed ... :ship: !

**Comment by rickybanister** [comment]
New copy sounds great, ship it

**Comment by klimeryk** [comment]
Just noting here that this broke renewals - fixed in https://github.com/Automattic/wp-calypso/pull/13984 by @dzver. That's because the `ManagePurchase` component was converted to a class, but that caused `handleRenew` to stop working because class methods are not bound by default. See https://facebook.github.io/react/docs/handling-events.html for more info.

**Pull request merged**